datafusion/expr/src/expr_fn.rs

```diff
@@ -434,10 +434,22 @@ impl SimpleScalarUDF {
         volatility: Volatility,
         fun: ScalarFunctionImplementation,
     ) -> Self {
-        let name = name.into();
-        let signature = Signature::exact(input_types, volatility);
-        Self {
+        Self::new_with_signature(
             name,
+            Signature::exact(input_types, volatility),
+            return_type,
+            fun,
+        )
+    }
+
+    pub fn new_with_signature(
```

Comment:
Could we please add some more documentation (specifically calling out the difference between this and `new`?

Reply:
good idea! added something, ptal